src/components/controls/InputContainer.tsx

```diff
@@ -67,6 +73,99 @@ const renderEditableContent = (props: IInputProps) => {
     );
 };
 
+const renderDatePicker = ({
+    value,
+    onChange,
+    isClearButtonVisible
+}: {
+    value?: Date;
+    isClearButtonVisible?: boolean;
+    onChange: (date: Date) => void;
+}) => {
+    const minDate = new Date();
+    minDate.setFullYear(1970);
+    minDate.setMonth(0);
+    minDate.setDate(1);
+    const maxDate = new Date();
+    maxDate.setFullYear(2051);
+    maxDate.setMonth(0);
+    maxDate.setDate(1);
+    // TODO: scroll to selected year missing from react-datepicker
+    // open issue for this: https://github.com/Hacker0x01/react-datepicker/pull/1700
+    return (
+        <div className={classnames(s.staticHeight)}>
+            <ReactDatePicker
+                customInput={renderDatePickerInput({
+                    value,
+                    onChange,
+                    isClearButtonVisible,
+                    placeholder: 'Syötä päivä'
+                })}
+                selected={value}
+                onChange={onChange}
+                locale={fi}
+                dateFormat={'dd.MM.yyyy'}
+                showMonthDropdown={true}
+                peekNextMonth={true}
+                showYearDropdown={true}
+                startDate={new Date()}
+                scrollableYearDropdown={true}
+                yearDropdownItemNumber={100}
+                minDate={minDate}
+                maxDate={maxDate}
+                dateFormatCalendar={'dd.MM.yyyy'}
+                popperContainer={_renderCalendarContainer}
+            />
+        </div>
+    );
+};
+
+const _renderCalendarContainer = ({ children }: { children: JSX.Element[] }): React.ReactNode => {
+    const el = document.getElementById('root');
+    return ReactDOM.createPortal(children, el!);
+};
+
+const renderDatePickerInput = ({
+    onChange,
+    placeholder,
+    value,
+    isClearButtonVisible
+}: {
+    onChange: (value: any) => void;
+    placeholder: string;
+    value?: Date;
+    isClearButtonVisible?: boolean;
+}) => {
+    const onInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+        // TODO: implement a better way of changing input value
+        const value = event.target.value;
+        if (Moment(value, 'DD.MM.YYYY', true).isValid()) {
```

Comment:
It don't seem to make any difference if lines 142-145 weren't there but then again it has TODO comment so I don't know if changes are necessary